ceph_bootstrap/deploy.py

```diff
@@ -1149,17 +1234,38 @@ def check_ceph_salt_formula():
                           "\"salt -G 'ceph-salt:member' saltutil.sync_all\" manually and fix "
                           "the problems reported")
                 return 1
-        logger.info("minions sync finished")
+
+        PP.println("Checking existing deployment...")
+        result = SaltClient.local().cmd('ceph-salt:member', 'ceph_orch.configured',
+                                        tgt_type='grain')
+        deployed = False
+        for minion, value in result.items():
+            if value:
+                deployed = len(SaltClient.local().cmd(minion, 'ceph_orch.host_ls')[minion]) > 0
+                break
+        # cluster already deployed
+        if minion_id is None and deployed:
+            logger.error("cluster already deployed and minion_id not provided")
+            PP.pl_red("Cluster is already deployed, please apply the deployment to a "
+                      "single minion at a time: \"ceph-bootstrap deploy <minion_id>\"")
+            return 1
+        # fist deployment
+        if minion_id is not None and not deployed:
+            logger.error("cluster not deployed and minion_id provided")
+            PP.pl_red("Cluster is not deployed yet, please apply the deployment to "
+                      "all minions at the same time: \"ceph-bootstrap deploy\"")
+            return 1
+
```

Comment:
We need a new validation that checks we are not applying the deployment on minions that are already "managed by orchestrator".

Reply:
Done